source/src/apps/public/analysis/SID_ERMS_Rescore.cc

```diff
@@ -0,0 +1,673 @@
+				if ( option[ B_vals ].user() ) {
+				//read in B values if given by file input
+					read_B_vals(B_);
+					TR << "B values read in from file" << std::endl;
+					if ( option[ basic::options::OptionKeys::in::file::l ].user() ) {
+					TR.Warning << "PDB file input not used. B values read from file instead. To calculate B from structure, remove the flag (B_vals)." << std::endl;
+					}
+				} else {
+					if ( !option[ basic::options::OptionKeys::in::file::l ].user() ) {
+						std::stringstream err_msg;
+						err_msg << "Must input either file containing B values (using flag B_vals) or PDB file to calculate B values for simulation." << std::endl;
+					utility_exit_with_message(err_msg.str());
+				}
+
+				calc_B_values(B_, edges_, A_, n_chains_, pose_in); //read in PDB and calculate B values for each the first in each interface type
+				TR << "B values read in from structure" << std::endl;
+			
+
+		//read in ERMS data: acceleration energies and ERMS data (if applicable)
+			utility::vector1<core::Real> ACE_; //Acceleration energies, read from file or input automatically
+			utility::vector1<utility::vector1<core::Real>> ERMS_;
+			if ( option[ ERMS ].user() ) {
+				read_ERMS(ACE_, ERMS_, n_chains_);
+				TR << "Acceleration energies and ERMS read in from file" << std::endl;
+			} else {
+			//if no acceleration energies are input, the range is set from 0 to 2x max B, with 10 steps in between
+				core::Real step_size = (*max_element(B_.begin(), B_.end()))*2/10;
+				ACE_.push_back(0.0);
+				for ( core::Size i=1; i<=10; i++ ) {
+					ACE_.push_back(i*step_size);
+				}
+				TR << "Acceleration energies set up based on max B" << std::endl;
+			}
+
+
+			//simulate ERMS
+			utility::vector1<utility::vector1<core::Real>> ERMS_prediction_(n_chains_);
+			simulate_ERMS(ERMS_prediction_, ACE_, B_, n_chains_, edges_, A_, breakage_cutoff_, nodes_);
+			TR << "ERMS simulation complete" << std::endl;
+
+			//calculte RMSE (if applicable)
+			if ( option[ RMSE ].user() && option[ ERMS ].user() ) {
+				calc_RMSE(ERMS_prediction_, ERMS_, n_chains_, ERMS_.size());
+			}
+			if ( option[ RMSE ].user() && !option[ ERMS ].user() ) {
+				TR.Warning << "RMSE cannot be calculated without ERMS data input." << std::endl;
+			}
+	
+			//get score of structure
+			core::Real score_ = sfxn->score( pose_in );
+
+			//output results to file
+			core::Real RMSE_value_(0.0); 
+			RMSE_value_ = calc_RMSE(ERMS_prediction_, ERMS_, n_chains_, ERMS_.size());
+			rmse_list.push_back(RMSE_value_);
+			score_list.push_back(score_);
+			}
+		}
+		//Determine maximum and minimum RMSE
+		core::Real max_rmse(0.0);
+		core::Real min_rmse(1.0);
+
+		for ( core::Size i = 1; i <= rmse_list.size(); i++ ) {
+        	if ( max_rmse < rmse_list[i] ) {
+            	max_rmse = rmse_list[i];
+        	}
+        }
+		for ( core::Size i = 1; i <= rmse_list.size(); i++ ) {
+        	if ( min_rmse > rmse_list[i] ) {
+            	min_rmse = rmse_list[i];
+        	}        
+        }
```

Comment:
If found it interesting you used std::max_element above, but you're not using it (or std::minmax_element) here. (That's fine, I just found it interesting.)